projects/openapi-utilities/src/openapi3/json-path-interpreters.ts

```diff
@@ -0,0 +1,54 @@
+import { jsonPointerHelpers } from '@useoptic/json-pointer-helpers';
+import { IFact } from './sdk/types';
+
+// A trie datastructure keyed by json path parts
+export type FactTree = Record<
+  string,
+  { fact: IFact | null; children: FactTree }
+>;
+
+// We need a special json path decoder since the leading / is significant and points to the root
+const decodePathWithLeadingTrail = (pointer: string) => pointer.split('/').map(jsonPointerHelpers.unescape);
+
+export function constructFactTree(facts: IFact[]): FactTree {
```

Comment:
This function can be adapted to just pass in list of json paths + locators instead of facts (when we remove facts at some point in the future).

This is going to be important as we lean more heavily on json path interpretations (I think this implementation is OpenAPI version agnostic, the version handling differences will be handled based on the facts that are passed in here).